utils/apollo/schema/resolvers.js

```diff
@@ -1,54 +1,130 @@
-import dbConnect from 'utils/dbConnect';
 import User from 'models/User';
 import Space from 'models/Spaces';
+
 export const resolvers = {
   Query: {
     users: async (parent, args, context) => {
       const { userIds, email, name } = args;
-      console.debug('args', args, 'userIds', userIds);
-      const data = [];
-
-      await dbConnect();
+      console.debug('args', args, 'userIds', userIds, 'context', context);
 
-      // Query by Name and Email
+      // Fetch only the current session user by Name and Email
       if (email && name) {
         try {
           const user = await User.findOne({ name, email });
-          data.push(user);
+          return [user];
         } catch (err) {
-          console.debug(err);
+          console.debug('Cannot fetch user', err);
         }
-        return data;
       }
 
       // For Query by ID
-      for (let id of userIds) {
-        try {
-          const user = await User.findById(id);
-          data.push(user);
-        } catch (err) {
-          console.debug(err);
-        }
+      const filter = {
+        _id: {
+          $in: userIds,
+        },
+      };
+
+      let users = [];
+      try {
+        users = await User.find(filter);
+        console.debug('users', users);
+      } catch (err) {
+        console.debug(err);
       }
-      return data;
+
+      return users;
     },
 
     spaces: async (parent, args, context) => {
       const { spaceIds } = args;
       console.debug('args', args, 'spaceIds', spaceIds);
-      const data = [];
 
-      await dbConnect();
+      // Fetch all spaces
+      if (spaceIds?.length == 0) {
+        const spaces = await Space.find({}).populate('participants').exec();
+        return spaces;
+      }
 
-      for (let spaceId of spaceIds) {
-        try {
-          const space = await Space.findOne({ spaceId });
-          data.push(space);
-        } catch (err) {
-          console.debug(err);
-        }
+      // For Query by ID
+      const filter = {
+        spaceId: {
+          $in: spaceIds,
+        },
+      };
+
+      let spaces = {};
+      try {
+        spaces = await Space.find(filter).populate('participants').exec();
+      } catch (err) {
+        console.debug(err);
+      }
+      return spaces;
+    },
+  },
+  Mutation: {
+    createSpace: async (parent, args, context) => {
+      console.debug('createSpace', args);
+      const { input } = args;
```

Comment:
you can probably just destructure args in the function parameters if you only needs input. Ex. `(parent, { input}, context) => ...`

Reply:
That's a good idea 🚀

Reply:
🚀